test/base_maths.jl

```diff
@@ -96,6 +97,13 @@ using Test
             @test A′ * b1 isa Matrix
             @test A′ * b1 ≈ A′ * Matrix(b1)
             @test_throws DimensionMismatch A * b1
+
+            # degenerate cases
+            m = rand(0, 0)
+            @test m * BlockDiagonal([m]) == m * m == m
+            m = rand(5, 0)
+            @test m' * BlockDiagonal([m]) == m' * m == rand(0, 0)
+            @test m * BlockDiagonal([m']) == m * m' == zeros(5, 5)
```

Comment:
I was thinking you could try a BlockDiagonal with no blocks too. But I don't think that really exists anywhere else here so I'm not going to ask for it now.

Reply:
actaully that's a great point - but needs a PR of its own I think
https://github.com/invenia/BlockDiagonals.jl/issues/30